[Feature] Compatibiltiy with `brainglobe-atlasapi` V3

The test suite fails when run with `brainglobe-atlasapi==3.0.0rc1` see https://github.com/brainglobe/brainglobe-atlasapi/issues/780.

There seem to be two key failures:

https://github.com/brainglobe/brainreg/blob/2d405316d0c5717bc66b65aabd63d88384ad5dea/brainreg/napari/register.py#L39-L48

Uses `utils.conf_from_file()` which now has a required cache path argument. This isn't really meant to be used outside of `brainglobe-atlasapi`. It's cleaner to replace this with a `brainglobe-atlasapi.list_atlases.get_all_atlases_lastversions()` as it does the same thing and also includes the custom atlases users may have.

The other error is more cryptic. Something related to logging it seems?
<details>

```
tests/tests/test_integration/test_registration.py:63:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
brainreg/core/cli.py:288: in main
    register(
brainreg/core/main.py:30: in main
    atlas = BrainGlobeAtlas(atlas)
            ^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/brainglobe_atlasapi/bg_atlas.py:119: in __init__
    self.check_latest_version()
../../miniforge3/envs/test-env/lib/python3.13/site-packages/brainglobe_atlasapi/bg_atlas.py:440: in check_latest_version
    remote_version = self.remote_version
                     ^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/brainglobe_atlasapi/bg_atlas.py:199: in remote_version
    if self.fs.exists(bucket_path) is False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/fsspec/asyn.py:118: in wrapper
    return sync(self.loop, func, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/fsspec/asyn.py:103: in sync
    raise return_result
../../miniforge3/envs/test-env/lib/python3.13/site-packages/fsspec/asyn.py:56: in _runner
    result[0] = await coro
                ^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/s3fs/core.py:1245: in _exists
    await self._info(path, bucket, key, version_id=version_id)
../../miniforge3/envs/test-env/lib/python3.13/site-packages/s3fs/core.py:1737: in _info
    out = await self._call_s3(
../../miniforge3/envs/test-env/lib/python3.13/site-packages/s3fs/core.py:464: in _call_s3
    return await _error_wrapper(
../../miniforge3/envs/test-env/lib/python3.13/site-packages/s3fs/core.py:215: in _error_wrapper
    raise err
../../miniforge3/envs/test-env/lib/python3.13/site-packages/s3fs/core.py:169: in _error_wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/aiobotocore/context.py:36: in wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/aiobotocore/client.py:386: in _make_api_call
    ) = await self._resolve_endpoint_ruleset(
../../miniforge3/envs/test-env/lib/python3.13/site-packages/aiobotocore/client.py:534: in _resolve_endpoint_ruleset
    endpoint_info = await self._ruleset_resolver.construct_endpoint(
../../miniforge3/envs/test-env/lib/python3.13/site-packages/aiobotocore/regions.py:29: in construct_endpoint
    LOG.debug(
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1508: in debug
    self._log(DEBUG, msg, args, **kwargs)
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1665: in _log
    self.handle(record)
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1681: in handle
    self.callHandlers(record)
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1737: in callHandlers
    hdlr.handle(record)
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1027: in handle
    self.emit(record)
../../miniforge3/envs/test-env/lib/python3.13/site-packages/_pytest/logging.py:399: in emit
    super().emit(record)
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1159: in emit
    self.handleError(record)
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:1151: in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:999: in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/site-packages/_pytest/logging.py:137: in format
    return super().format(record)
           ^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/test-env/lib/python3.13/logging/__init__.py:712: in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <LogRecord: aiobotocore.regions, 10, /home/igor/miniforge3/envs/test-env/lib/python3.13/site-packages/aiobotocore/regions.py, 29, "Calling endpoint provider with parameters: %s">

    def getMessage(self):
        """
        Return the message for this LogRecord.

        Return the message for this LogRecord after merging any user-supplied
        arguments with the message.
        """
        msg = str(self.msg)
        if self.args:
>           msg = msg % self.args
                  ^^^^^^^^^^^^^^^
E           TypeError: not all arguments converted during string formatting
```

</details>

--- a/brainreg/napari/register.py
+++ b/brainreg/napari/register.py
@@ -9,7 +9,7 @@ import brainglobe_space as bg
 import napari
 import numpy as np
 from brainglobe_atlasapi import BrainGlobeAtlas
-from brainglobe_atlasapi.list_atlases import descriptors, utils
+from brainglobe_atlasapi.list_atlases import get_all_atlases_lastversions
 from brainglobe_napari_io.brainmapper.brainmapper_reader_dir import (
     load_registration,
 )
@@ -41,11 +41,7 @@ def get_available_atlases():
     Get the available brainglobe atlases
     :return: Dict of available atlases (["name":version])
     """
-    available_atlases = utils.conf_from_url(
-        descriptors.remote_url_base.format("last_versions.conf")
-    )
-    available_atlases = dict(available_atlases["atlases"])
-    return available_atlases
+    return get_all_atlases_lastversions()
 
 
 def add_registered_image_layers(

--- a/tests/test_brainreg_napari.py
+++ b/tests/test_brainreg_napari.py
@@ -55,6 +55,9 @@ def test_workflow(make_napari_viewer, tmp_path):
         pixel_widget = getattr(widget, f"{dim}_pixel_um")
         pixel_widget.value = brain_layer.metadata["voxel_size"][i]
 
+    # Set atlas to example_mouse_100um
+    widget.atlas_key.value = widget.atlas_key.annotation["example_mouse_100um"]
+
     assert len(viewer.layers) == 1
 
     # Run registration

